pyk/src/pyk/cterm/cterm.py

```diff
@@ -322,6 +313,23 @@ def from_dict(dct: dict[str, Any]) -> CSubst:
         constraints = (KInner.from_dict(c) for c in dct['constraints'])
         return CSubst(subst=subst, constraints=constraints)
 
+    @staticmethod
+    def from_pred(pred: KInner) -> CSubst:
+        """Extract from a boolean predicate a CSubst."""
+        subst, pred = extract_subst(pred)
+        return CSubst(subst=subst, constraints=flatten_label('#And', pred))
+
+    def pred(self, sort_with: KDefinition | None = None, subst: bool = True, constraints: bool = True) -> KInner:
+        """Return an ML predicate representing this substitution."""
+        _preds: list[KInner] = []
+        if subst:
+            for k, v in self.subst.minimize().items():
+                sort = K if not (sort_with and sort_with.sort(v)) else not_none(sort_with.sort(v))
```

Comment:
Here, `sort_with.sort` is called twice if `sort_with` and `sort_with.sort(v)`.

Reply:
Done, unfortunately it's more verbose, maybe there is a better way.

Reply:
Looks good to me.